File: iOS/TodoList/Model/TodoItem.swift
//
//  TodoItem.swift
//  TodoList
//
//  Created by Morgan Pritchard on 2/26/26.
//
import Foundation

enum ItemStatus: String, CaseIterable {
    case notStarted = "Not Started"
    case inProgress = "In Progress"
    case completed = "Completed"
    case archived = "Archived"
}

struct TodoItem : Identifiable {
    let id = UUID().uuidString
    let title: String
    let desc: String?
    let itemStatus: ItemStatus
    let creationDate: Date = Date()
    let lastUpdated: Date = Date()
}

let mockData: [TodoItem] = [
    
    // MARK: - Not Started
    TodoItem(
        title: "Buy groceries",
        desc: "Milk, Eggs, Bread, Spinach",
        itemStatus: .notStarted
    ),
    
    TodoItem(
        title: "Schedule dentist appointment",
        desc: nil,
        itemStatus: .notStarted
    ),
    
    
    // MARK: - In Progress
    TodoItem(
        title: "Build SwiftUI login screen",
        desc: "Implement validation and error handling",
        itemStatus: .inProgress
    ),
    
    TodoItem(
        title: "Read 20 pages of Clean Code",
        desc: "Focus on functions chapter",
        itemStatus: .inProgress
    ),
    
    
    // MARK: - Completed
    TodoItem(
        title: "Morning workout",
        desc: "30 min strength training",
        itemStatus: .completed
    ),
    
    TodoItem(
        title: "Submit project report",
        desc: nil,
        itemStatus: .completed
    ),
    
    
    // MARK: - Archived
    TodoItem(
        title: "2024 Tax Filing",
        desc: "All documents submitted and approved",
        itemStatus: .archived
    ),
    
    TodoItem(
        title: "Old portfolio redesign",
        desc: "Replaced with new 2025 version",
        itemStatus: .archived
    )
]

File: web/app/model/todoitem.ts
export enum ItemStatus {
  NOT_STARTED = "Not Started",
  IN_PROGRESS = "In Progress",
  COMPLETED = "Completed",
  ARCHIVED = "Archived",
}

export interface TodoItem {
  id: string;
  title: string;
  desc?: string;
  itemStatus: ItemStatus;
  creationDate: Date;
  lastUpdated: Date;
}

export const mockData: TodoItem[] = [
  // MARK: - Not Started
  {
    id: crypto.randomUUID(),
    title: "Buy groceries",
    desc: "Milk, Eggs, Bread, Spinach",
    itemStatus: ItemStatus.NOT_STARTED,
    creationDate: new Date(),
    lastUpdated: new Date(),
  },
  {
    id: crypto.randomUUID(),
    title: "Schedule dentist appointment",
    itemStatus: ItemStatus.NOT_STARTED,
    creationDate: new Date(),
    lastUpdated: new Date(),
  },

  // MARK: - In Progress
  {
    id: crypto.randomUUID(),
    title: "Build SwiftUI login screen",
    desc: "Implement validation and error handling",
    itemStatus: ItemStatus.IN_PROGRESS,
    creationDate: new Date(),
    lastUpdated: new Date(),
  },
  {
    id: crypto.randomUUID(),
    title: "Read 20 pages of Clean Code",
    desc: "Focus on functions chapter",
    itemStatus: ItemStatus.IN_PROGRESS,
    creationDate: new Date(),
    lastUpdated: new Date(),
  },

  // MARK: - Completed
  {
    id: crypto.randomUUID(),
    title: "Morning workout",
    desc: "30 min strength training",
    itemStatus: ItemStatus.COMPLETED,
    creationDate: new Date(),
    lastUpdated: new Date(),
  },
  {
    id: crypto.randomUUID(),
    title: "Submit project report",
    itemStatus: ItemStatus.COMPLETED,
    creationDate: new Date(),
    lastUpdated: new Date(),
  },

  // MARK: - Archived
  {
    id: crypto.randomUUID(),
    title: "2024 Tax Filing",
    desc: "All documents submitted and approved",
    itemStatus: ItemStatus.ARCHIVED,
    creationDate: new Date(),
    lastUpdated: new Date(),
  },
  {
    id: crypto.randomUUID(),
    title: "Old portfolio redesign",
    desc: "Replaced with new 2025 version",
    itemStatus: ItemStatus.ARCHIVED,
    creationDate: new Date(),
    lastUpdated: new Date(),
  },
];